.github/workflows/build-java.yaml

```diff
@@ -0,0 +1,102 @@
+#
+# Just a workflow demonstrating how to build and release prql-java.
+# It should be eventually merged into nightly and release workflows.
+#
+name: build-java
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "prql-java/**"
+      - ".github/workflows/build-java.yaml"
+  workflow_call:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-java
+  cancel-in-progress: true
+
+jobs:
+  build-jni:
+    strategy:
+      matrix:
+        # start with the most popular ones
+        include:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+            suffix: ".dylib"
+            jdk: 8
+          - os: macos-latest
+            target: aarch64-apple-darwin
+            suffix: ".dylib"
+            # https://adoptium.net/supported-platforms/
+            jdk: 11
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+            suffix: ".so"
+            jdk: 8
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            suffix: ".dll"
+            jdk: 8
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install JDK 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: ${{ matrix.jdk }}
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args:
+            --release --manifest-path
+            prql-java/prql-api/src/main/rust/Cargo.toml
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.target }}
+          path: target/release/libprql_java${{ matrix.suffix }}
+          if-no-files-found: error
+
+  build-java:
```

Comment:
At some point we should probably split this up into a "Test" and "Release" workflow, which could each call the "Build" workflow.

Reply:
Right. But I hope to make this one work first :)